### DISCUSSION TITLE
My logo he doesn't bump

### DISCUSSION THREAD

**Discussion by DeadlyneDev** [YoshiCrafter29/YoshiCrafterEngine#206] [Q&A]
![image](https://user-images.githubusercontent.com/101023010/169354875-86dd5bbc-be3e-4efc-9af6-39024712bb25.png)

my logo doesen't bump pls help me:
the script:
import("LoadingState");

var crafterEngineLogo:FlxSprite = null;
var BackgroudTilte:FlxSprite;

function create() {

    crafterEngineLogo = new FlxSprite( -20, -35);
    crafterEngineLogo.frames = Paths.getSparrowAtlas('titlescreen/logoSUS');
    crafterEngineLogo.antialiasing = true;
    crafterEngineLogo.animation.addByPrefix('bump', 'logo bumpin', 5);
    crafterEngineLogo.animation.play('bump');
    crafterEngineLogo.updateHitbox();
    crafterEngineLogo.scale.x = crafterEngineLogo.scale.y = 0.95;
    crafterEngineLogo.screenCenter();
    add(crafterEngineLogo);

    BackgroudTilte = new FlxSprite().loadGraphic(Paths.image("menuBG"));
	BackgroudTilte.screenCenter();
	BackgroudTilte.antialiasing = ClientPrefs.globalAntialiasing;
    add(BackgroudTilte);
}

var danced = true;
function beatHit() {
    crafterEngineLogo.animation.play(danced ? "bump" : "bump");
    danced = !danced;
}

function update(elapsed:Float) {
    if (FlxG.mouse.justPressed) {
        var pos = FlxG.mouse.getScreenPosition();
        if (pos.x >= 9 && pos.x < 186 && pos.y >= 238 && pos.y < 411) {
            // gray you fucking genius
            CoolUtil.loadSong("YoshiCrafterEngine", "yoshi", "normal");
            LoadingState.loadAndSwitchState(new PlayState_());
        }
    }
    }

**Comment by BernardoGP4504** [comment]
everything's wrong here. The script for it should be:
```
var Logo:FlxSprite;
var bg:FlxSprite;

function create(){
    bg = new FlxSprite(0, 0).loadGraphic(Paths.image('menuBG'));
    bg.screenCenter(); //use this line if only you want to center the bg
    add(bg);

    Logo = new FlxSprite(0, 0);
    Logo.x -= 20;
    Logo.y -= 35;
    Logo.screenCenter();
    Logo.scale.x = Logo.scale.y = 0.95;
    Logo.frames = Paths.getSparrowAtlas('titlescreen/logoSUS');
    Logo.antialiasing = true;
    Logo.animation.addByPrefix('bump', 'logo bumpin', 24, true); //change 'logo bumpin' right here if it does not match your xml 
    add(Logo);
}

function beatHit() {
    Logo.animation.play('bump');
}
```

**Reply by BernardoGP4504** [reply]
if there's an error that says something like: "unexpected token" and it's talking about Logo then change Logo into logo
